src/app/store/reducers/VehicleTestResultModel.reducers.spec.ts

```diff
@@ -53,5 +57,53 @@ describe('VehicleTestResultModel Reducer', () => {
     });
   });
 
-});
+  describe('[SetCurrentState]', () => {
+    test('should update currentState', () => {
+      const action = new SetCurrentState(VIEW_STATE.VIEW_ONLY);
+      const result = VehicleTestResultModelReducers(initialVehicleTestResultModelState, action);
+
+      expect(result).toMatchObject({
+        ...initialVehicleTestResultModelState,
+        editState: VIEW_STATE.VIEW_ONLY
+      });
+    });
+  });
+
+  describe('[Update Test Result]', () => {
+    test('should update testStations', () => {
+      const vehicleTestResultTestTypeNumber: TestResultTestTypeNumber = {
+        testResultUpdated: {} as TestResultModel,
+        testTypeNumber: '1'
+      } as TestResultTestTypeNumber;
 
+      initialVehicleTestResultModelState.vehicleTestResultModel = [
+        { testTypes: [{ testNumber: '1' }] } as TestResultModel,
+        { testTypes: [{ testNumber: '2' }] } as TestResultModel
+      ];
+
+      const action = new UpdateTestResult(vehicleTestResultTestTypeNumber);
+
+      const updatedVehicleTestResults = initialVehicleTestResultModelState.vehicleTestResultModel.map(
```

Comment:
I will need to have an understanding of this please.
I am confused, it is not showing a simple
 arrange act assert pattern
We already defined the initial state shape a few lines above